packages/compass-indexes/src/modules/search-indexes.ts

```diff
@@ -510,24 +515,41 @@ export const fetchSearchIndexes = (): IndexesThunkAction<
       return;
     }
 
-    if (status !== SearchIndexesStatuses.PENDING) {
-      // 2nd time onwards set the status to refreshing
-      dispatch(setIsRefreshing());
+    // If we are currently doing fetching indexes, we will
+    // wait for that
+    if (NOT_FETCHABLE_STATUSES.includes(status)) {
+      return;
     }
 
     try {
+      dispatch({ type: ActionTypes.SetStatus, status: newStatus });
       const indexes = await dataService.getSearchIndexes(namespace);
-      const sortedIndexes = _sortIndexes(indexes, sortColumn, sortOrder);
-      dispatch(setSearchIndexes(sortedIndexes));
+      dispatch(setSearchIndexes(_sortIndexes(indexes, sortColumn, sortOrder)));
     } catch (err) {
-      dispatch(setError((err as Error).message));
+      // We do no set any error on poll or refresh and the
+      // previous list of indexes is shown to the user.
+      if (newStatus === 'FETCHING') {
+        dispatch(setError((err as Error).message));
+      }
     }
   };
 };
 
-export const refreshSearchIndexes = (): IndexesThunkAction<void> => {
+export const refreshSearchIndexes = (): IndexesThunkAction<Promise<void>> => {
+  return async (dispatch, getState) => {
+    const { indexes } = getState().searchIndexes;
+
+    // If we have a list of indexes already, then we are refreshing
+    // or else we are fetching them.
+    const newStatus: SearchIndexesStatus =
+      indexes.length > 0 ? 'REFRESHING' : 'FETCHING';
```

Comment:
Wondering about the common case of having no search indexes yet. Then it will be FETCHING every time and what we probably want is REFRESHING? Wondering if we should base this off the previous status rather.

Reply:
I added another status to keep track of that, `NOT_READY` and in that state we have not fetched anything yet. once we do that, we do not go back to that state again.

Reply:
What is FETCHING, then? Can't it just be NOT_READY?

Reply:
Seems like you can only transition from NOT_READY to FETCHING or from READY to REFRESHING. then from FETCHING or REFRESHING to READY. or READY to POLLING and back to READY.

Reply:
Yes, from NOT_READY to FETCHING once. And FETCHING is like the initial loading state where we probably want to show centered spinner. 

And to READY state, we can get from all the states that fetch data (FETCHING, REFRESHING or POLLING).